src/field/generated_fieldops.cxx

```diff
@@ -0,0 +1,1668 @@
+// This file is autogenerated - see gen_fieldops.py
+#include <bout/mesh.hxx>
+#include <field2d.hxx>
+#include <field3d.hxx>
+#include <globals.hxx>
+#include <interpolation.hxx>
+
+// Do the actual multiplication of Field3D and Field3D
+void autogen_Field3D_Field3D_Field3D_multiplication(BoutReal *__restrict__ result,
+                                                    const BoutReal *__restrict__ lhs,
+                                                    const BoutReal *__restrict__ rhs,
+                                                    int len) {
+
+  for (int i = 0; i < len; ++i) {
+    result[i] = lhs[i] * rhs[i];
+  }
+}
+
+// Provide the C++ wrapper for multiplication of Field3D and Field3D
+Field3D operator*(const Field3D &lhs, const Field3D &rhs) {
+  Indices i{0, 0, 0};
+  Mesh *localmesh = lhs.getMesh();
+
+  ASSERT1(localmesh == rhs.getMesh());
+
+  Field3D result(localmesh);
+  result.allocate();
+  checkData(lhs);
+  checkData(rhs);
+
+  autogen_Field3D_Field3D_Field3D_multiplication(&result[i], &lhs[i], &rhs[i],
+                                                 localmesh->LocalNx * localmesh->LocalNy *
+                                                     localmesh->LocalNz);
+
+#if CHECK > 0
```

Comment:
Shouldn't this check come _before_ we do the operation?

Reply:
Yes, I've moved it to top of each operator